provider/provider_plugin_framework.go

```diff
@@ -0,0 +1,177 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/databricks/databricks-sdk-go/client"
+	"github.com/databricks/databricks-sdk-go/config"
+	"github.com/databricks/databricks-sdk-go/useragent"
+	"github.com/databricks/terraform-provider-databricks/commands"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/databricks/terraform-provider-databricks/pluginframework"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/provider"
+	"github.com/hashicorp/terraform-plugin-framework/provider/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+var pluginFrameworkProviderName = "databricks-tf-provider-pluginframework"
+
+func init() {
+	// IMPORTANT: this line cannot be changed, because it's used for
+	// internal purposes at Databricks.
+	useragent.WithProduct(pluginFrameworkProviderName, common.Version())
+}
+
+func GetDatabricksProviderPluginFramework() provider.Provider {
+	p := &DatabricksProviderPluginFramework{}
+	return p
+}
+
+type DatabricksProviderPluginFramework struct {
+}
+
+var _ provider.Provider = (*DatabricksProviderPluginFramework)(nil)
+
+func (p *DatabricksProviderPluginFramework) Resources(ctx context.Context) []func() resource.Resource {
+	return []func() resource.Resource{
+		pluginframework.ResourceQualityMonitor(),
+	}
+}
+
+func (p *DatabricksProviderPluginFramework) DataSources(ctx context.Context) []func() datasource.DataSource {
+	return []func() datasource.DataSource{
+		pluginframework.DataSourceVolumes(),
+	}
+}
+
+func (p *DatabricksProviderPluginFramework) Schema(ctx context.Context, req provider.SchemaRequest, resp *provider.SchemaResponse) {
+	resp.Schema = providerSchemaPluginFramework()
+}
+
+func (p *DatabricksProviderPluginFramework) Metadata(ctx context.Context, req provider.MetadataRequest, resp *provider.MetadataResponse) {
+	resp.TypeName = pluginFrameworkProviderName
+	resp.Version = common.Version()
+}
+
+func (p *DatabricksProviderPluginFramework) Configure(ctx context.Context, req provider.ConfigureRequest, resp *provider.ConfigureResponse) {
+	client := configureDatabricksClient_PluginFramework(ctx, req, resp)
+	resp.DataSourceData = client
+	resp.ResourceData = client
```

Comment:
How can we override the DatabricksClient used for tests? One idea is to use a custom version of the provider for testing:
```go
type DatabricksProviderPluginFramework struct {
    client *common.DatabricksClient
}

func (p *DatabricksProviderPluginFramework) Configure(...) {
    if p.client == nil {
        p.client = configureDatabricksClient_PluginFramework(...)
    }
    resp.DataSourceData = p.client
    resp.ResourceData = p.client
}
```
Then in unit tests when we want to mock the workspace client, we can do it here.